diff --git a/mocktracer/BUILD b/mocktracer/BUILD
index 3b22bab..607aa39 100644
--- a/mocktracer/BUILD
+++ b/mocktracer/BUILD
@@ -11,9 +11,14 @@ cc_library(
 
 cc_binary(
     name = "libmocktracer_plugin.so",
+    linkopts = [
+        "-Wl,--version-script=$(location :tracer-export-map.ld)",
+    ],
     linkshared = 1,
+    linkstatic = 1,
     visibility = ["//visibility:public"],
     deps = [
-        "//mocktracer:mocktracer"
+        "//mocktracer",
+        "//mocktracer:tracer-export-map.ld",
     ],
 )
diff --git a/mocktracer/tracer-export-map.ld b/mocktracer/tracer-export-map.ld
new file mode 100644
index 0000000..66a4a4f
--- /dev/null
+++ b/mocktracer/tracer-export-map.ld
@@ -0,0 +1,5 @@
+{
+  global:
+    OpenTracingMakeTracerFactory;
+  local: *;
+};
