--- BUILD.bazel.orig	2019-08-07 14:51:34.179907100 -0400
+++ BUILD.bazel	2019-08-09 11:00:02.013270500 -0400
@@ -54,11 +54,11 @@
 # Use #pragma GCC diagnostic ignored in integration code to suppress these errors.
 quiche_copt = [
     # Remove these after upstream fix.
-    "-Wno-unused-parameter",
-    "-Wno-unused-function",
-    "-Wno-type-limits",
-    # quic_inlined_frame.h uses offsetof() to optimize memory usage in frames.
-    "-Wno-invalid-offsetof",
+    # "-Wno-unused-parameter",
+    # "-Wno-unused-function",
+    # "-Wno-type-limits",
+    # # quic_inlined_frame.h uses offsetof() to optimize memory usage in frames.
+    # "-Wno-invalid-offsetof",
 ]
 
 envoy_cc_test_library(
